### PR TITLE
delete player

### PR DESCRIPTION
**Pull request by Sarahtonein**
Created delete player endpoint, opening pr change

### CODE REVIEW

File: controllers/rest/player/DeletePlayer.go
Comment (Balugrizzly) [inline controllers/rest/player/DeletePlayer.go:1]: Not mandatory and could be smthing for later but would be nice to have 

- Name, UserID support

**Review by Balugrizzly — CHANGES_REQUESTED** [review]
You forgot:
- Changelog

And the Documentation is not 100% accurate just FYI but thats minor and up to you if you want to address that

File: controllers/rest/player/DeletePlayer.go
Comment (Sarahtonein) [reply controllers/rest/player/DeletePlayer.go:1]: > Not mandatory and could be smthing for later but would be nice to have
> 
> * Name, UserID support

I have added Name and UserID, using query params for all three options (ID, Name, UserID)

**Comment by Sarahtonein** [comment]
> You forgot:
> 
> * Changelog
> 
> And the Documentation is not 100% accurate just FYI but thats minor and up to you if you want to address that

Documentation has been amended to address changes.

File: controllers/rest/player/DeletePlayer.go
Comment (Balugrizzly) [inline controllers/rest/player/DeletePlayer.go:1]: You could have used a single path param and bound that into a interface or a generic.
Which is often also directly supported downstream by like gorm or whatever 
just FYI

File: controllers/rest/player/DeletePlayer.go [hunk 1]
```diff
@@ -14,27 +15,52 @@ func DeletePlayer(c echo.Context) error {
...
 
-    if idToRemove == "" {
-        return c.JSON(http.StatusBadRequest, "Invalid 'id' parameter value. Use a valid ID.")
+    queryPlayer := &models.Player{}
+
+    //check ID
+    if idParam != "" {
+        idUint, err := strconv.ParseUint(idParam, 10, 64)
+        if err != nil {
+            c.Logger().Error("Invalid 'id' parameter value. Use a valid numeric ID.")
```
Comment (Balugrizzly) [inline controllers/rest/player/DeletePlayer.go:28]: This allows Log fludding and ddosing behaviour 
consider changing it to a debug function or removing it entirly

File: controllers/rest/player/DeletePlayer.go [hunk 2]
```diff
@@ -14,27 +15,52 @@ func DeletePlayer(c echo.Context) error {
...
+    // Check Name
+    if nameParam != "" {
+        queryPlayer.Name = nameParam
+    }
 
+    // Check UserID
+    if userIDParam != "" {
+        userID, err := uuid.FromString(userIDParam)
+        if err != nil {
+            c.Logger().Error("Invalid 'userid' parameter value. Use a valid UUID.")
```
Comment (Balugrizzly) [inline controllers/rest/player/DeletePlayer.go:43]: Same as above

**Review by Balugrizzly — CHANGES_REQUESTED** [review]
Apart from the Minor points looks good and can be merged after those changes

File: controllers/rest/player/DeletePlayer.go
Comment (Sarahtonein) [reply controllers/rest/player/DeletePlayer.go:1]: > You could have used a single path param and bound that into a interface or a generic. Which is often also directly supported downstream by like gorm or whatever just FYI


I gave this a go but not sure if this has overcomplicated things.

Comment (Balugrizzly) [reply controllers/rest/player/DeletePlayer.go:1]: This is absolutely not what i meant :)

**Review by Balugrizzly — CHANGES_REQUESTED** [review]
Please revise

File: controllers/rest/player/DeletePlayer.go (hunk 1 above)
Comment (Sarahtonein) [reply controllers/rest/player/DeletePlayer.go]: > This allows Log fludding and ddosing behaviour consider changing it to a debug function or removing it entirly

I am not so sure I have resolved this with current changes, I feel like we also need to also implement things such as rate limiting if we want to reduce DDOS like behavior

**Comment by Sarahtonein** [comment]
> Please revise

Revision has been done, to be discussed :)

File: controllers/rest/player/DeletePlayer.go
Comment (Balugrizzly) [reply controllers/rest/player/DeletePlayer.go:1]: What are you doing?
I think you are going way of path here

**Review by Balugrizzly — COMMENTED** [review]
What are you doing?
I think you are going way of path here

**Comment by Balugrizzly** [comment]
Lets talk about all this in voice

File: controllers/rest/player/DeletePlayer.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    var arg interface{}
+
+    if _, err := uuid.FromString(identifier); err == nil {
+        query, arg = "user_id = ?", identifier
+    } else if _, err := strconv.Atoi(identifier); err == nil {
+        id, _ := strconv.Atoi(identifier)
+        query, arg = "id = ?", id
+    } else {
+        query, arg = "name = ?", identifier
+    }
```
Comment (Balugrizzly) [inline controllers/rest/player/DeletePlayer.go:28]: You do not need this.
You can set the values in the query player e.g

**Review by Balugrizzly — COMMENTED** [review]
Looks Good 
There is the one comment i left.
Up to you if you want to change or merge it like this

File: controllers/rest/player/DeletePlayer.go (hunk 3 above)
Comment (Sarahtonein) [reply controllers/rest/player/DeletePlayer.go]: > You do not need this. You can set the values in the query player e.g

I've done this through queryPlayer instead and tested. If all good I will update the README/Changelog if applicable.

File: README.md [hunk 4]
```diff
@@ -389,28 +389,50 @@ http://127.0.0.1:1323/player?active=true
 
 ##### DeletePlayer
 
-`DELETE /player`
+`DELETE /player{identifier}`
```
Comment (Balugrizzly) [inline README.md:392]: you forgot the / and its usually /:id

**Review by Balugrizzly — COMMENTED** [review]
Apart from the tiny missing / if you say this is ready i will merge

File: controllers/rest/player/DeletePlayer.go (hunk 3 above)
Comment (Balugrizzly) [reply controllers/rest/player/DeletePlayer.go]: not 100% what i meant but all good :) 
please update readme and changelog and let me know when its ready

**Comment by Sarahtonein** [comment]
> Apart from the tiny missing / if you say this is ready i will merge

This has been updated :)

**Review by Balugrizzly — APPROVED** [review]
This Looks good

**Pull request merged**